drivers/soundwire/intel.c

```diff
@@ -1056,11 +1033,39 @@ static void *intel_get_sdw_stream(struct snd_soc_dai *dai,
 	return dma->stream;
 }
 
+static int intel_trigger(struct snd_pcm_substream *substream, int cmd, struct snd_soc_dai *dai)
+{
+	struct sdw_cdns *cdns = snd_soc_dai_get_drvdata(dai);
+	struct sdw_intel *sdw = cdns_to_intel(cdns);
+	struct sdw_cdns_dma_data *dma;
+
+	/*
+	 * The .prepare callback is used to deal with xruns and resume operations. In the case
+	 * of xruns, the DMAs and SHIM registers cannot be touched, but for resume operations the
+	 * DMAs and SHIM registers need to be initialized.
+	 * the .trigger callback is used to track the suspend case only.
+	 */
+	if (cmd != SNDRV_PCM_TRIGGER_SUSPEND)
```

Comment:
This might work if:
```
if (cmd != SNDRV_PCM_TRIGGER_SUSPEND || cmd != SNDRV_PCM_TRIGGER_STOP)
	return 0;
```
But I;m not 100% sure if the prepare callback is going to be executed before `SNDRV_PCM_TRIGGER_RESUME`
The `SNDRV_PCM_TRIGGER_STOP` might be part of the xrun recovery, so that will likely cause another set of issues...

Reply:
no, on STOP we will eventually go to HW_FREE to release the resources. If you do a STOP/START it's fine to keep the values as is.